go/caddy/pfpki/cloud/intune.go

```diff
@@ -195,22 +183,27 @@ func (cl *Intune) NewCloud(ctx context.Context, name string) error {
 		if k == "value" {
 			for _, n := range v.([]interface{}) {
 				for a, b := range n.(map[string]interface{}) {
-					if a == "serviceName" {
+					if a == "providerName" {
 						if b == VALIDATION_SERVICE_NAME {
-							apiEndpoint.OdataType = n.(map[string]interface{})["odata.type"].(string)
-							apiEndpoint.ObjectId = n.(map[string]interface{})["objectId"].(string)
-							apiEndpoint.ResourceId = n.(map[string]interface{})["resourceId"].(string)
-							apiEndpoint.ObjectType = n.(map[string]interface{})["objectType"].(string)
-							apiEndpoint.Capability = n.(map[string]interface{})["capability"].(string)
-							apiEndpoint.ServiceId = n.(map[string]interface{})["serviceId"].(string)
-							apiEndpoint.ServiceName = n.(map[string]interface{})["serviceName"].(string)
 							apiEndpoint.Uri = n.(map[string]interface{})["uri"].(string)
 						}
 					}
 				}
 			}
 		}
 	}
+
+	// Fetch the token for intune api
+	tk, err = cred.GetToken(
+		context.TODO(), policy.TokenRequestOptions{Scopes: []string{intuneResourceUrl + "/.default"}},
+	)
+
+	if err == nil {
+		cl.AccessToken = "Bearer " + tk.Token
+	} else {
+		log.Print(err)
```

Comment:
Should we return this error?

Reply:
Yes, for troubleshooting purpose it´s really usefull.